tokio/src/fs/file.rs

```diff
@@ -837,6 +854,19 @@ mod preadv2 {
         }
     }
 
+    #[cfg_attr(test, allow(unused))]
+    pub(crate) fn has_buggy_preadv2() -> bool {
+        let mut uts: MaybeUninit<libc::utsname> = MaybeUninit::uninit();
+        let res = unsafe { libc::uname(uts.as_mut_ptr()) };
+        if res < 0 {
+            // Getting kernel version failed, assume that it's buggy
+            return true;
+        }
+        let release = unsafe { CStr::from_ptr((*uts.as_ptr()).release.as_ptr()) };
+        let release = release.to_bytes();
+        release[..4] == b"5.9."[..] || release[..5] == b"5.10."[..]
```

Comment:
You beat me to it :)

FWIW in the impl I was brewing I made this comparison with `[i8]` avoiding the conversion via `CStr` like this:

```rust
        const _5_9: [i8; 4] = [53, 46, 57, 46]; // [i8] repr of b"5.9."
        const _5_10: [i8; 5] = [53, 46, 49, 48, 46]; // [i8] repr of b"5.10."
        uts.release[..4] == _5_9 || uts.release[..5] == _5_10
```

Not that it matters, this function will only be called ~once per process.